sklearn/model_selection/tests/test_validation.py

```diff
@@ -769,3 +771,22 @@ def test_cross_val_predict_with_method():
         predictions = cross_val_predict(est, X, y, method=method,
                                         cv=kfold)
         assert_array_almost_equal(expected_predictions, predictions)
+
+
+def test_score_memmap():
+    # Ensure a scalar score of memmap type is accepted
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    clf = MockClassifier()
+    tf = tempfile.NamedTemporaryFile(mode='wb', delete=False)
+    tf.write('Hello world!!!!!')
+    tf.close()
+    scores = np.memmap(tf.name, dtype=float)
+    score = scores.sum()
```

Comment:
Btw, I was hoping that was not permanent numpy behaviour, but it is because of that behaviour that this is an issue for us.